### ISSUE TITLE
[ISSUE] Destroying databricks_workspace_conf fails when using storeInteractiveNotebookResultsInCustomerAccount

### ISSUE THREAD

**Issue by laurensknoll** [databrickslabs/terraform-provider-databricks#823]
Hi,

Databricks currently allows us to store all interactive notebook results in the Azure/AWS account. Also see https://docs.databricks.com/administration-guide/workspace/storage.html#configure-the-storage-location-for-all-interactive-notebook-results

When this feature is configured using the `databricks_workspace_conf`-resource. The feature is enabled accordingly after deployment. On destroy, however, the provider fails with the following error: `Some values are not allowed: {"storeInteractiveNotebookResultsInCustomerAccount":""}`

Apparently, the provider sends out a empty string value instead of false value.

### Configuration
```hcl
resource "databricks_workspace_conf" "this" {
  custom_config = {
    "storeInteractiveNotebookResultsInCustomerAccount" : true
  }
}
```

### Expected Behavior
On destroy, the databricks workspace configuration should be destroyed without any error. 

### Actual Behavior
The destroy fails, due to a bad request:  `Some values are not allowed {"storeInteractiveNotebookResultsInCustomerAccount":""}`

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. `terraform destroy`

### Terraform and provider versions
Terraform v1.0.2
AzureRM 2.61
Databricks v0.3.6

**Comment by alexott** [comment]
Fixed by #826

**Comment by SudoSpartanDan** [comment]
@alexott Seems this issue exists still with some attributes:

```
Error: Some values are not allowed: {"maxTokenLifetimeDays":""}
```

This gets passed in as a string and the API for the configurations doesn't seems to like that it's just a blank string.

https://github.com/databrickslabs/terraform-provider-databricks/blob/57d334d0173ba7b70ae95de068df0a6958fc2f49/workspace/resource_workspace_conf.go#L104-L118

**Comment by alexott** [comment]
why are you passing empty value?

**Comment by SudoSpartanDan** [comment]
I’m not, this is on destruction.

**Comment by SudoSpartanDan** [comment]
I think the issue is the API for this particular value expects ```"0"``` rather than just a blank string, so on destruction it fails.


https://docs.databricks.com/dev-tools/api/latest/token-management.html#operation/set-configuration

**Comment by alexott** [comment]
yep, I'll look later today or tmorrow morning

**Comment by abij** [comment]
Running into the same issue. On destruction with a non boolean setting in the workspace: The setting is actually: "1".
- `Terraform v1.3.4`
- `databricks/databricks v1.3.0` 

`Error: cannot delete workspace conf: Some values are not allowed: {"maxTokenLifetimeDays":"false"}`

**Comment by abij** [comment]
Related (closed) issue: #1229.

In my case the lifeTime is `1`, therefor it _can_ be parsed as a bool and becomes `"false"` instead of empty :D.
I'll set the value to `2` instead of `1`, which is not parsed as a bool: https://pkg.go.dev/strconv#ParseBool
